chainstate/src/detail/mod.rs

```diff
@@ -164,11 +164,24 @@ impl<S: BlockchainStorage, V: TransactionVerificationStrategy> Chainstate<S, V>
             time_getter,
         );
 
```

Comment:
To avoid having to store a separate genesis block id, simply do the following:

instead of that if condition below, match on it.

- If it's None, then the genesis block isn't initialized. No problem there. Just run the normal initialization.
- If it's Some, check the best height:
    - If the height is > 0, retrieve block number 1, check the prev block id. If it's genesis, you've proven what you wanted to prove.
    - If the height is 0, the current best block should be genesis block.

Reply:
I started with this but then thought it would be conceptually simpler to just store the genesis ID. Can do what you suggest, I don't have a strong preference either way

Reply:
Would be great if you could go through with it. Just isolate it in a separate function `check_genesis_matches` or something. I have many fears that relate to people abusing the value of genesis being in storage in the future instead of properly using `ChainConfig`.